parlai/mturk/core/dev/agents.py

```diff
@@ -177,9 +104,31 @@ class MTurkAgent(Agent):
     def __init__(self, opt, mturk_manager, hit_id, assignment_id, worker_id):
         super().__init__(opt)
 
-        self.conversation_id = None
-        self.mturk_manager = mturk_manager
+        # all MTurkManager functions explicitly used by agents extracted here
```

Comment:
I wanted `mturk_manager` dependencies to all be explicit rather requiring the full context of a manager in the agent state. (This way it's more obviously going to break if we make a change in that class that breaks here too.)